src/components/AvatarCropModal/ImageCropView.js

```diff
@@ -52,17 +52,24 @@ const defaultProps = {
 const ImageCropView = (props) => {
     const containerStyle = StyleUtils.getWidthAndHeightStyle(props.containerSize, props.containerSize);
 
+    const originalImageHeight = props.originalImageHeight;
```

Comment:
Why bother reassigning everything here instead of just referencing props?

Reply:
I guess this might be related to https://github.com/software-mansion/react-native-reanimated/pull/4060

Assume that you use `obj.prop` in a worklet. Previously, there was an optimization in the Reanimated Babel plugin which captures a new object `{ prop: prop }` which contains only `prop` instead of the original `obj` object which can have more properties. This worked well for `ref.current`, `props.foo` etc. but we had to get rid of this optimization in https://github.com/software-mansion/react-native-reanimated/pull/4060 because it turned out that it broke other things. Unfortunately, when we removed this optimization, the whole `obj` was copied to the worklet runtime and this didn't work well for objects with cyclic references (in particular, for React refs) so we had to implement a debug mode-only check in https://github.com/software-mansion/react-native-reanimated/pull/4367 to detect such cases and throw an error in debug mode.

We'll definitely need to revisit this topic in the future but at this point (since Reanimated 3.0.0) this is actually the recommended way of passing things to worklets.

Reply:
@tomekzaw this copy-only-values never properly worked for me and I didn't even know it existed, to be honest. From the early beginning of Reanimated I knew you should never use full objects directly because of some random problems/crashes. Sometimes it might work, sometimes it didn't. 

So better never try and just use values directly.

Reply:
Cool, thanks for the explanation ❤️